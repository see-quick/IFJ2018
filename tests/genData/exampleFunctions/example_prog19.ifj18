#Taylor polynom for log

def taylor_log(x, n)
  result = 0.0
	x_exp = 1
	numerator_fraction = 0.0
	i = 1

	if x > 1 then
		numerator_fraction = (x - 1) / x

		while i <= n do
			x_exp = x_exp * numerator_fraction
			result = result + x_exp / i
			i = i + 1
		end
	else
		x = 1 - x

		while i <= n do
			x_exp = x_exp * x
			result = result - (x_exp / i)
			i = i + 1
		end
	end
	result = result
end

	d = inputi
	print "\n"
	d = taylor_log(d, 10)
	print d, "\n"

	d = inputi
	print "\n"
	d = taylor_log(d, 20)
	print d, "\n"
