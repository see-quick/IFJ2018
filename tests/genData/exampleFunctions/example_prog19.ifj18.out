5
0x1.65df657b0cf14p-1
10
0x1.8988cb7af117ap-1
